src/soundsourceproxy.cpp

```diff
@@ -433,9 +433,69 @@ QImage SoundSourceProxy::parseCoverImage() const {
     return coverImg;
 }
 
+namespace {
+
+// Keeps the TIO alive while accessing the audio data
+// of the track. The TIO must not be deleted while
+// accessing the corresponding file to avoid file
+// corruption when writing metadata while the file
+// is still in use.
+class AudioSourceProxy: public Mixxx::AudioSource {
+public:
+    AudioSourceProxy(const AudioSourceProxy&) = delete;
+    AudioSourceProxy(AudioSourceProxy&&) = delete;
+
+    static Mixxx::AudioSourcePointer create(
+            const TrackPointer& pTrack,
+            const Mixxx::AudioSourcePointer& pAudioSource) {
+        DEBUG_ASSERT(!pTrack.isNull());
+        DEBUG_ASSERT(!pAudioSource.isNull());
+        return Mixxx::AudioSourcePointer(
+                new AudioSourceProxy(pTrack, pAudioSource));
+    }
+
+    AudioSourceProxy(
+            const TrackPointer& pTrack,
+            const Mixxx::AudioSourcePointer& pAudioSource)
+        : Mixxx::AudioSource(*pAudioSource),
+          m_pTrack(std::move(pTrack)),
```

Comment:
From my understanding using std::move() on a const reference requires to create a temporary copy. In this case that is no difference to passing the parameter by value in the first place and using std::move() to efficiently extract its contents.

Using an r-value ref as a parameter type would put the burden to supply an r-value ref on the caller, even if it wants to keep the supplied to parameter. It saves one move construction, but leaks implementation details outside of the function.

Simple rules:
- Use r-value reference parameters only for move construction/assignment
- Caller: If you pass a value to as a parameter function that you don't access afterwards, release ownership by using std::move() to enable move construction for this parameter
- Callee: Use std::move() to read parameters that are passed by value for initializing members or when consuming a parameter value and do not access this parameter in any other way